Gordon360/Services/RecIM/MatchService.cs

```diff
@@ -0,0 +1,411 @@
+                                           }),
+                            })
+                        }).FirstOrDefault();
+            return match;
+        }
+
+        public IEnumerable<TeamMatchHistoryViewModel> GetMatchHistoryByTeamID(int teamID)
+        {
+            var vm = _context.Match
+                            .Join(_context.MatchTeam
+                                .Where(mt => mt.TeamID == teamID)
+
+                                    .Join(
+                                        _context.MatchTeam.Where(mt => mt.TeamID != teamID),
+                                        mt0 => mt0.MatchID,
+                                        mt1 => mt1.MatchID,
+                                        (mt0, mt1) => new
+                                        {
+                                            TeamID = mt0.TeamID,
+                                            MatchID = mt0.MatchID,
+                                            Score = mt0.Score,
+                                            OpposingID = mt1.TeamID,
+                                            OpposingTeamScore = mt1.Score,
+                                            Status = mt0.Score > mt1.Score
+                                                    ? "Win"
+                                                    : mt0.Score < mt1.Score
+                                                        ? "Lose"
+                                                        : "Tie"
+                                        }),
+
+
+                                match => match.ID,
+                                matchTeamJoin => matchTeamJoin.MatchID,
+                                (match, matchTeamJoin) => new TeamMatchHistoryViewModel
+                                {
+                                    TeamID = matchTeamJoin.TeamID,
+                                    MatchID = match.ID,
+                                    Opponent = _context.Team.Where(t => t.ID == matchTeamJoin.OpposingID)
+                                        .Select(o => new TeamExtendedViewModel
+                                        {
+                                            ID = o.ID,
+                                            Name = o.Name,
+                                            Logo = o.Logo
+                                        }).FirstOrDefault(),
+                                    TeamScore = matchTeamJoin.Score,
+                                    OpposingTeamScore = matchTeamJoin.OpposingTeamScore,
+                                    Status = matchTeamJoin.Status,
+                                    MatchStatusID = match.StatusID,
+                                    Time = match.Time
+                                }).AsEnumerable();
+            return vm;
+        }
+        public IEnumerable<MatchExtendedViewModel> GetMatchBySeriesID(int seriesID)
+        {
+            var match = _context.Match
```

Comment:
This should probably be plural since it's a collection of matches:
```suggestion
        public IEnumerable<MatchExtendedViewModel> GetMatchesBySeriesID(int seriesID)
        {
            var matches = _context.Match
```